test/libdrampowertest/lib_test.cc

```diff
@@ -82,6 +82,21 @@ int main(int argc, char* argv[])
         // Need at least tWRAPDEN = AL + CWL + BL/2 + WR + 1 cycles between WR and PDN_F_PRE
         test.doCommand(MemCommand::PDN_F_PRE,3,265);
 
+        // Exit from Precharge Power-down
+        test.doCommand(MemCommand::PUP_PRE,3,300);
+        // Activate bank 0
+        test.doCommand(MemCommand::ACT,0,350);
+        // Precharge all banks with bank 0 active
+        test.doCommand(MemCommand::PREA,0,400);
+        // Precharge all banks again
+        test.doCommand(MemCommand::PREA,0,450);
+        // Activate bank 0 twice
```

Comment:
Actually, the generation of warnings is not tested here. The lines added will exercise more parts of the code (the inside of some "if" statements) . By doing this the test coverage is increased and we know that the program will not crash in the situations explored.
The warnings are indeed generated, but the python script that calls the library test program compares only the standard output (excluding warnings).